libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -224,7 +224,7 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 				else if (sourceBaseType->isValueType())
 					CompilerUtils(_context).loadFromMemoryDynamic(*sourceBaseType, fromCalldata, true, false);
 				else
-					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage not yet supported.");
+					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage not yet supported in legacy codegen. Note: It is implemented in via IR.");
```

Comment:
How about this?

```suggestion
					solUnimplemented("Copying of type " + _sourceType.toString(false) + " to storage is only supported by the IR pipeline. Hint: try compiling with `--via-ir` (CLI) or the equivalent `viaIR: true` (Standard JSON)");
```

Reply:
Much better. Although, this is an error for unimplemented features. Shouldn't the primary focus of the error message be to describe where it's not implemented i.e. legacy? and then as a side note include that they can try it with via ir?

Reply:
Well, in this case if it's not one then it's the other so not much difference IMO. Just thought it sounded a bit better this way. But if you prefer, I'm fine with your version for the first sentence too. TBH I mostly wanted to fix the weirdly sounding "in via IR" bit and also list the flags like we do in the StackTooDeep message :)

Reply:
How about now?